lib/linter.js

```diff
@@ -262,8 +262,8 @@ class Linter {
     return currentSource;
   }
 
-  async updateTodo(linterOptions, results, todoConfig) {
-    await this._todoHandler.update(linterOptions.filePath, results, todoConfig);
+  updateTodo(linterOptions, results, todoConfig) {
+    return this._todoHandler.update(linterOptions.filePath, results, todoConfig);
```

Comment:
I'd like to revert these changes and keep this declared as an async function. It makes it much easier to spot on quick review that it is returning a promise.

Reply:
It would basically be `return await` I guess?

Reply:
Mentioned above, if this was driven by linting we should fix the linting.

Reply:
It was driven by linting. One thing to note is that `processTodos` below also returns a promise, but the signature matches this change. So, either I think we do both one way, or the other. I'm assuming you'd prefer to `return await` to better signify a promise being returned, correct. 

Also important to note, if we used TS we'd be able to mark the return type and avoid this back and forth :)

Reply:
We removed the no-return-await bit in https://github.com/ember-template-lint/ember-template-lint/pull/1689 so I think we can restructure this back to an async function now.

Reply:
Done.